stdlib/public/core/Sequence.swift

```diff
@@ -605,11 +605,11 @@ public protocol Sequence {
   /// in the same order.
   func _copyToContiguousArray() -> ContiguousArray<Iterator.Element>
 
-  /// Copy a Sequence into an array, returning one past the last
-  /// element initialized.
-  @discardableResult
-  func _copyContents(initializing ptr: UnsafeMutablePointer<Iterator.Element>)
-    -> UnsafeMutablePointer<Iterator.Element>
+  /// Copy `self` into an unsafe buffer, returning a partially-consumed 
+  /// iterator with any elements that didn't fit remaining.
```

Comment:
The reason I made the iterator null is because generating an empty iterator (i.e. an iterator that has had all its elements consumed) might be non-trivial for optimized implementations like Array._copyContents that don't actually use an iterator for their copy – because when you are inside your own _copyContents, you know *you* didn't lie about your size, so you can just use a pointer. So it's a pain for `Array` to have to create an `IndexingIterator` then manually set its position to `_endIndex`. Dave raised similar concerns so I'm going to try taking that back out and see what the code is like.